app/src/update_db.py

```diff
@@ -78,3 +83,112 @@ def update_database_legacy(region, ladder_list):
+    realm, profile_id, ladder_id = None, None, None
+    if ladder:
+        realm, profile_id, ladder_id = ladder.realm, ladder.profile_id, ladder.ladder_id
+
+    api_as = SC2PublicAPI(BASE_URL, client=uplink.AiohttpClient())
+    crawler = LadderCrawler(api_as, max_requests=512)
+    loop = asyncio.get_event_loop()
+
+    # TODO: Can filter out recently updated ladders or players
+    #  ladder_ids = [ladder.ladder_id for ladder in Ladder.objects.filter(ladder_filter)]
+    #  player_filter = Q(rank=league_dict[league], region=Region.EU.value)
+    #  profile_ids = [str(player.profile_id) for player in Player.players.filter(player_filter)]
+    #  crawler.ladder_ids = ladder_ids
+    #  crawler.profile_ids = profile_ids
+
+    ladder_infos = loop.run_until_complete(crawler.get_adjacent_ladders(
+                                                region=region,
+                                                league=league,
+                                                realm=realm,
+                                                profile_id=profile_id,
+                                                ladder_id=ladder_id
+                                                                        )
+                                            )
+    return ladder_infos
+
+
+def update_database_public_ladders(ladder_infos):
+    for ladder in ladder_infos:
+        if ladder is None:
+            continue
+        league_dict = {"grandmaster": Rank.GRANDMASTER.value, "master": Rank.MASTER.value,
+                       "diamond": Rank.DIAMOND.value, "platinum": Rank.PLATINUM.value, "gold": Rank.GOLD.value,
+                       "silver": Rank.SILVER.value, "bronze": Rank.BRONZE.value}
+        Ladder.objects.update_or_create(ladder_id=ladder.ladder_id,
+                                        region=ladder.p.region,
+                                        defaults={'realm': ladder.p.realm,
+                                                  'profile_id': ladder.p.profile_id,
+                                                  'rank': league_dict[ladder.league]
+                                                  }
+                                        )
+
+
+def update_public_ladders(region, league=None):
+    region_dict = {'us': 1, 'eu': 2, 'kr': 3}
+    region = region_dict.get(region)
+    if region is None:
+        for r in range(1, 4):
+            ladders = fetch_public_ladders(region=r, league=league)
+            update_database_public_ladders(ladders)
+    else:
+        ladders = fetch_public_ladders(region=region, league=league)
+        update_database_public_ladders(ladders)
+
+
+def fetch_public_players(region, league=None):
+    api_as = SC2PublicAPI(BASE_URL, client=uplink.AiohttpClient())
+    crawler = LadderCrawler(api_as, max_requests=12)
+    loop = asyncio.get_event_loop()
+    bundle_list = list()
+    league_dict = {"grandmaster": Rank.GRANDMASTER.value, "master": Rank.MASTER.value,
```

Comment:
probably want to store this somewhere common since it's in like 3 places, but yea